File: api-code-examples/rust/blobs-list-blobs.rs
use futures_lite::StreamExt;

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let node = iroh::node::Node::memory().spawn().await?;

    // Define content and tag
    let content = "hello world!";

    // Add blob
    let outcome = node.blobs().add_bytes(content).await?;
    println!("Added blob {} ({} bytes)", outcome.hash, outcome.size);

    println!("blobs list:");

    // List blobs
    let mut blobs = node.blobs().list().await?;
    while let Some(info) = blobs.next().await {
        let info = info?;
        println!("\t{}", info.hash);
    }

    Ok(())
}

File: api-code-examples/nodejs/blobs-list-blobs.js
import { Iroh } from '@number0/iroh'

const node = await Iroh.memory()

// Define content and tage
const content = Array.from(Buffer.from("hello world!"))

// Add blob
const outcome = await node.blobs.addBytes(content)
console.log(`Added blob ${outcome.hash} (${outcome.size} bytes)`)

console.log('blobs list:');

// List blobs
const blobs = await node.blobs.list()
for (let blob of blobs) {
  console.log(`\t${blob}`)
}
